orbital-camera.ts

```diff
@@ -0,0 +1,253 @@
+import {Component} from '@wonderlandengine/api';
+import {property} from '@wonderlandengine/api/decorators.js';
+import {deg2rad} from './utils/utils.js';
+
+const preventDefault = (e: Event) => {
+    e.preventDefault();
+};
+
+const tempVec = [0, 0, 0];
+
+/**
+ * OrbitalCamera component allows the user to orbit around a target point, which
+ * is the position of the object itself. It rotates at the specified distance.
+ *  
+ * @remarks
+ * The component works using mouse or touch. Therefor it does not work in VR.
+ */
+export class OrbitalCamera extends Component {
+    static TypeName = 'orbital-camera';
+
+    @property.int()
+    mouseButtonIndex = 0;
+
+    @property.float(5)
+    radial = 5;
+
+    @property.float()
+    minElevation = 0.0;
+
+    @property.float(89.99)
+    maxElevation = 89.99;
+
+    @property.float()
+    minZoom = 0.01;
+
+    @property.float(10.0)
+    maxZoom = 10.0;
+
+    @property.float(0.5)
+    xSensitivity = 0.5;
+
+    @property.float(0.5)
+    ySensitivity = 0.5;
+
+    @property.float(0.02)
+    zoomSensitivity = 0.02;
+
+    @property.float(0.9)
+    decelerationFactor = 0.9;
+
+    private mouseDown: boolean = false;
+    private origin = [0, 0, 0];
+    private azimuth = 0;
+    private polar = 45;
+
+    private initialPinchDistance: number = 0;
+    private touchStartX: number = 0;
+    private touchStartY: number = 0;
+
+    private azimuthSpeed: number = 0;
+    private polarSpeed: number = 0;
+
+    start(): void {
+        this.object.getPositionWorld(this.origin);
+        this.updateCamera();
+    }
+
+    onActivate(): void {
+        const canvas = this.engine.canvas;
+
+        canvas.addEventListener('mousemove', this.onMouseMove);
+        if (this.mouseButtonIndex === 2) {
+            canvas.addEventListener('contextmenu', preventDefault, {passive: false});
+        }
+        canvas.addEventListener('mousedown', this.onMouseDown);
+        canvas.addEventListener('mouseup', this.onMouseUp);
+        canvas.addEventListener('wheel', this.onMouseScroll, {passive: false});
+
+        canvas.addEventListener('touchstart', this.onTouchStart, {passive: false});
```

Comment:
Wouldn't it be bettee to use the pointer API instead? So it's somehow unified mouse+touch just with an extra pointer